AmplifyPlugins/Auth/AWSAuthPlugin/Dependency/AuthorizationProviderAdapter+SignedInSession.swift

```diff
@@ -0,0 +1,194 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import AWSCore
+import AWSMobileClient
+import AWSPluginsCore
+
+extension AuthorizationProviderAdapter {
+
+    /// Creates session for a signedIn user.
+    ///
+    /// Session values available as a signedIn user depend on the way the user signedIn and the configurations enabled
+    /// for the current signedIn user. The following scenarios can happen:
+    /// - User is signedIn through Cognito User Pool and Identity Pool is not configured
+    /// - User is signedIn through Cognito User Pool but refresh token is expired
+    /// - User is signedIn through Cognito User Pool but access token couldnot be refreshed because of N/W or
+    /// service issue
+    /// - User is signedIn through Cognito User Pool but aws credentials couldnot be refreshed because of N/W or
+    /// service issue
+    /// - User is signedIn through Cognito Identity Pool
+    ///- User is signedIn through Cognito Identity Pool but aws credentials couldnot be refreshed because of N/W or
+    ///service issue
+    /// - Parameter completionHandler: Completion handler to return the result.
+    func fetchSignedInSession( _ completionHandler: @escaping (Result<AuthSession, AmplifyAuthError>) -> Void) {
+
+        // User can be signed in to user pools or federated to identityPool.
+        // Call getTokens first to verify that the signedIn user is in UserPool or not.
+        awsMobileClient.getTokens { [weak self] tokens, error in
+            guard let self = self else {
+                return
+            }
+            guard error == nil else {
+
+                if let urlError = error as NSError?, urlError.domain == NSURLErrorDomain {
+                    let authSession = AuthCognitoSignedInSessionHelper.offlineSignedInSession()
+                    completionHandler(.success(authSession))
```

Comment:
See if you can break the handlers for these various error conditions into separate methods, to help with the readability of this method. Right now, it's very hard to follow why you make the decisions you do, and some well-named methods would go a long way toward making your intent clearer. Ideally you'd be cascading these decisions in easy-to-consume chunks, although please do balance the practical considerations of dealing with underlying AWSMobileClient patterns against the need for clarity in this internal module--don't go to extraordinary lengths to clean up code if it means weird contortions to pass values through various callbacks.

Reply:
Moved few logics to different methods.